apps/site/pages/docs/api-extensions.mdx

```diff
@@ -1,463 +1,127 @@
-        </td>
-      </ExtendableQueryTable>
+      <summary><b>Extendable queries</b></summary>
+        <br />
+        Extendable queries in FastStore's GraphQL API are predefined GraphQL queries that serve as a foundation for retrieving data from the API. 
+        These queries allows you to add or modify fields within the query to tailor the data retrieval process to the store specific needs.
+        <br />
     </details>
   </li>
 </ul>
 
 ---
 
-## How to consume FastStore API Extension
-
-
-Once you define which new fields will be exposed by the FastStore API Extension, then determine the GraphQL fragments where these fields will be used.
-Finally, it's time to know how to consume the new fields in your [New Sections](/docs/building-sections/creating-a-new-section) or [Custom Overridable Components](/docs/building-sections/overriding-components-and-props).
+## Best practices for fetching data
+Site performance tends to get worse as more API requests are made by a page. Because of this, the best practices for fetching data on your storefront involve getting precisely the data you need with the minimum number possible of requests to the FastStore API. The sections below go over what you can do to minimize the API requests of your pages.
 
-It's important to know that FastStore exposes the data that comes from FastStore API along with FastStore API Extensions inside a provider called `PageProvider`.
-This data comes as a `context` in `PageProvider`, and you can use some hooks to access it:
+### Fetch only the data you need
+GraphQL allows you to customize your query to get precisely the data you need, which reduces the data returned from the requests. Double-check your queries to make sure they are not over-fetching.
 
-The `usePDP()` hook is recommendable in cases where you use your section on PDP.
+This also means that you must be mindful of the data brought into your query by [GraphQL fragments](https://graphql.org/learn/queries/#fragments). Use them only when they do not include unnecessary data.
 
-```ts copy
-import { usePDP } from "@faststore/core"
 
-const context = usePDP()
-```
+### Do not send requests to other APIs
 
-The `usePLP()` hook is recommendable in cases where you use your section on PLP.
+When building your storefront with FastStore, to avoid comprimising the website performance and lead to `504` timeout errors, do not send requests to APIs other than the FastStore API.
+If you need to access other data not available in the native FastStore API schema, you must do this by using [API extension](/docs/api-extensions). *See the code's comment*
 
-```ts copy
-import { usePLP } from "@faststore/core"
+{/*WILL E CONTINUE WITH THIS RECOMMENDATION? If you need data that is not available on the native [FastStore GraphQL schema](https://v1.faststore.dev/reference/api/objects), check the [root object](https://v1.faststore.dev/reference/api/objects).
```

Comment:
> If you need data that is not available on the native [FastStore GraphQL schema](https://v1.faststore.dev/reference/api/objects), check the [root object](https://v1.faststore.dev/reference/api/objects).

I think we should keep this topic as you've written in lines 83 and 84, but I don't know if the quoted text makes much sense now. I still think we should talk about `If you need data that is not available on the native [FastStore GraphQL schema]`, but I don't know if this is addressed in the `/docs/api-extensions` docs linked above. If it is, then it's safe to remove it from here.